docs/samples/Microsoft.ML.Samples/Dynamic/LogisticRegression.cs

```diff
@@ -63,7 +63,7 @@ public static void Example()
                 .Append(ml.Transforms.Text.FeaturizeText("TextFeatures", "Text"))
                 .Append(ml.Transforms.Concatenate("Features", "TextFeatures", "age", "fnlwgt",
                     "education-num", "capital-gain", "capital-loss", "hours-per-week"))
-                .Append(ml.BinaryClassification.Trainers.LogisticRegression());
+                .Append(ml.BinaryClassification.Trainers.LbfgsCalibrated());
```

Comment:
>LbfgsCalibrated [](start = 57, length = 15)

I don't think we should call it this. `LogisticRegression` is fine but `LbfgsCalibrated` is too esoteric.

Reply:
Plus it's not a calibrated linear model. It's fitting the to the logistic loss function.

Reply:
As long as it returns `CalibratedModelParametersBase<LinearBinaryModelParameters, PlattCalibrator`, I feel `LbfgsCalibrated` is fine. SDCA is doing the same.

Reply:
Technically true, but wouldn't it be clearer from a user perspective to call it `LogisticRegression`? Everybody knows what that is. Behind the scenes, many packages solve LR with variants of L-BFGS (e.g. Spark), so I don't think it would be misleading to say that it's called `LogisticRegression`. We can put technical details like "Implemented with L-BFGS etc." in the docs.

Look at `FastTree`. We don't call that `CalibratedLambdaMART` ;)

---
In reply to: [267857743](https://github.com/dotnet/machinelearning/pull/3034#discussion_r267857743) [](ancestors = 267857743)

Reply:
A problem here is that L-BFGS is not good for large-scale sparse data sets. SDCA in that case is usually much faster. If we call LBFGS-LogisticRegression just LogisticRegression, external users may view LBFGS as the default solver to all logistic regression problems.